custom_components/switchbotremote/client/client.py

```diff
@@ -46,24 +51,48 @@ def request(self, method: str, path: str, **kwargs) -> Any:
 
         if response.status_code != 200:
             _LOGGER.debug(f"Received http error {response.status_code} {response.text}")
-            raise RuntimeError(f"SwitchBot API server returns status {response.status_code}")
+            if response.status_code != 500:
+                raise HomeAssistantError(f"SwitchBot API server returns status {response.status_code}")
+            else:
+                raise SwitchbotInternal500Error
 
         response_in_json = humps.decamelize(response.json())
         if response_in_json["status_code"] != 100:
             _LOGGER.debug(f"Received error in response {response_in_json}")
-            raise RuntimeError(f'An error occurred: {response_in_json["message"]}')
+            raise HomeAssistantError(f'An error occurred: {response_in_json["message"]}')
 
         _LOGGER.debug(f"Call service {url} OK")
         return response_in_json
+    
+    def filtered_request(self, method: str, path: str, **kwargs) -> Any:
```

Comment:
@MrXANA91 I don't like to have unused code, what about the request api? Is it still used? 

In case not, change request api instead of add a new one, and add a parameter for "retry" that could be MAX_TRRIES as default and with 0 you should have the same behavior of the old request implementation.

Reply:
If you check line 73, you can see a call to the request API. `filtered_request` just wraps the current `request` method to catch any 500 status that could potentially appear.

First request is sent (considered first try). If a 500 status is caught and `MAX_TRIES` is greater than 1, the request is resent after waiting 500ms (or the value of `DELAY_BETWEEN_TRIES_MS`). If every requests throw a 500 error code, a total number of `MAX_TRIES` (here is 5) requests will be sent.

If you set the value of `MAX_TRIES` to 1, this code will behave exactly like it currently does in the release.

Reply:
mmmm ok, good point. Anyway, I don't like the name, why it is filtered? 

Could you use for:else (https://www.w3schools.com/python/gloss_python_for_else.asp) returning the result instead of breaking the loop?

Reply:
Anyway, I prefer to make request "private" -> so you could name your method request and I would like to accept the retry as parameter that per default it is equals to MAX_RETRIES

Reply:
Got it, I'll make the changes.

Do you want the delay value as a parameter as well?

Reply:
Yup, that's great.

Thanks a lot for your help and passion!